controllers/nodeagent.go

```diff
@@ -187,38 +218,57 @@ func (r *DPAReconciler) buildResticDaemonset(dpa *oadpv1alpha1.DataProtectionApp
 	var err error
 	ds.Labels, err = common.AppendUniqueKeyTOfTMaps(ds.Labels, installDs.Labels)
 	if err != nil {
-		return nil, fmt.Errorf("restic daemonset label: %s", err)
+		return nil, fmt.Errorf("NodeAgent daemonset label: %s", err)
 	}
 	// Update Spec
 	ds.Spec = installDs.Spec
 	ds.Name = dsName
 
-	return r.customizeResticDaemonset(dpa, ds)
+	return r.customizeNodeAgentDaemonset(dpa, ds)
 }
 
-func (r *DPAReconciler) customizeResticDaemonset(dpa *oadpv1alpha1.DataProtectionApplication, ds *appsv1.DaemonSet) (*appsv1.DaemonSet, error) {
-	if dpa.Spec.Configuration.Restic == nil {
-		// if restic is not configured, therefore not enabled, return early.
+func (r *DPAReconciler) customizeNodeAgentDaemonset(dpa *oadpv1alpha1.DataProtectionApplication, ds *appsv1.DaemonSet) (*appsv1.DaemonSet, error) {
+	if dpa.Spec.Configuration == nil || (dpa.Spec.Configuration.Restic == nil && dpa.Spec.Configuration.NodeAgent == nil) {
```

Comment:
Shouldn't this be `dpa.Spec.Configuration == nil || (dpa.Spec.Configuration.Restic == nil || dpa.Spec.Configuration.NodeAgent == nil)`

Reply:
@shubham-pampattiwar Your test will always be true, since other validation prevents both Restic and NodeAgent from being configured. It's one or the other. So "no node agent" means that "either the whole configuration is nil or *both* restic and node agent are unconfigured"

Reply:
So I think the above change is correct.

Reply:
It's correct. Applying De Morgan's law, we could have different:

```
A or B == not (A and B)
A and B == not A or not B
```

```go
if dpa.Spec.Configuration == nil || (dpa.Spec.Configuration.Restic == nil && dpa.Spec.Configuration.NodeAgent == nil) {}
```
equals to:

```go
if !(dpa.Spec.Configuration != nil && (dpa.Spec.Configuration.Restic != nil || dpa.Spec.Configuration.NodeAgent != nil)){}
```

In here we are checking if the following is not configured (bail out quick)
```
dpa.Spec.Configuration
```

Or if the `dpa.Spec.Configuration` is configured then we need to bail out when both `Restic` *and* `NodeAgent` are not configured, this means we don't apply logic.

Checking if both Restic and NodeAgent are present at the same time is done in other place with this PR:
https://github.com/openshift/oadp-operator/blob/7c07be18ba9ece757c00ff9c480a4c83e000a407/controllers/validator.go#L26-L28

Reply:
> De Morgan's law

sending me back to college I see.